### PR TITLE
Add support for all missing Regex character classes

### PR DESCRIPTION
**Pull request by CogentRedTester**
## Adds support for all missing JavaScript Regex character classes

- [ ] This PR  only introduces changes to documentation.
  - Please include a summary of changes and an explanation
- [x] This PR adds new functionality
  - [x] Is there a related issue?
    - Please note the related issue below, and how this PR relates to the issue if appropriate
  - [x] Does the code style reasonably match the existing code?
  - [x] Are the changes tested (using the existing format, as far as is possible?)
  - [x] Are the changes documented in the readme with a suitable example?
  - [x] Is the table of contents updated?
  - [x] Is the `index.d.ts` file updated, using appropriate types and using the same description as the readme?
- [ ] This PR introduces some other kind of change
  - Please explain the change below

This PR adds support for all remaining character classes as specified in the [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Character_classes). This PR builds on top of PR #67. May partially address #62.

The new additions are:

**.verticalTab** - `\v`.

**.formFeed** - `\f`.

**.backspace** - `\b` (evaluates to `[\b]` when not fused).

**.controlChar(X)** - `\cX` where `X` is a one digit letter from `A-Z`.

**.hexCode(HH)** - `\xHH` where `HH` is a 2 digit hexadecimal string.

**.utf16Code(HHHH)** - `\uHHHH` where `HHHH` is a 4 digit hexadecimal string.

**.unicodeCharCode(HHHHH)** - `\u{HHHH}` where `HHHHH` is a 4 or 5 digit hexadecimal string. Enables the `u` flag automatically.

**.unicodeProperty(property)** - `\p{property}` where `property` is a Unicode Character Property. Enables the `u` flag automatically.

**.notUnicodeProperty(property)** - `\P{property}` where `property` is a Unicode Character Property. Enables the `u` flag automatically.

All methods which take input are checking that the input is the correct type, length, and consists of the correct characters, and are throwing errors correctly when the input is invalid. The `.unicodeProperties()` function attempts to compile the given property into a RegExp object to test if the property is valid. If this is not done the error will only be thrown when the `.toRegex()` method is called, better I think to throw immediately. I can also split this into multiple PRs if you prefer.

Feel free to dictate any changes to method names, documentation, or behaviour.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -293,6 +293,23 @@ declare class SuperExpressive {
...
+     */
+    unicodeCharCode(hex: string): SuperExpressive;
+
+    /**
+     * Matches a unicode character with given `property` value.
+     * Invalid Unicode properties or values will cause .toRegex() to throw an error.
+     * Enables the unicode `u` flag when used.
+     * @param property A Unicode character property in the form `loneProperty` or `property=value`.
+     * For valid properties see the MDN Docs:
+     * https://developer.mozilla.org/docs/Web/JavaScript/Reference/Regular_expressions/Unicode_character_class_escape
```
Comment (francisrstokes) [inline index.d.ts:309]: This indeed seems like the sensible way to solve this 👍

File: index.test.js [hunk 2]
```diff
@@ -295,6 +295,33 @@ describe('SuperExpressive', () => {
...
+    `For valid properties see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Unicode_character_class_escape`,
+    () => SuperExpressive().unicodeProperty('~')
+  )
+  testErrorConditition(
+    'unicodeProperty: invalid property name',
+    `Property is not a valid Unicode property (got nope). ` +
+    `For valid properties see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Unicode_character_class_escape`,
+    () => SuperExpressive().unicodeProperty('nope')
+  )
+  // SuperExpressive().unicodeProperty('a').toRegex();
```
Comment (francisrstokes) [inline index.test.js:324]: I guess this can go

**Review by francisrstokes — CHANGES_REQUESTED** [review]

File: index.test.js (hunk 2 above)
Comment (CogentRedTester) [reply index.test.js:324]: Oh woops, a commented out line snuck through. Is that the only thing you want me to remove?

Comment (francisrstokes) [reply index.test.js:324]: Yeah just the comment. The other tests look good. I wasn't going to nitpick on it, but if you're removing the line, could you also add the missing semicolons in a few places :smile:

**Comment by CogentRedTester** [comment]
@francisrstokes the changes have been made

**Comment by CogentRedTester** [comment]
Actually don't merge yet, I just realised I missed something.

**Comment by CogentRedTester** [comment]
I have now added support for the `\P` code, which is the inverse of the `\p` code. I have named the method `notUnicodeProperty`. `nonUnicodeProperty` matched the naming convention, but I didn't think it sounded right. `withoutUnicodeProperty` may be an alternative.

**Pull request merged**